File: GO/ex2.go
package main

import "fmt"

func main() {
	var n int
	fmt.Print("Digite um número inteiro positivo: ")
	fmt.Scan(&n)

	if n <= 0 {
		fmt.Println("n não é compatível")
		return
	}

	isPrimo := true
	if n == 1 {
		isPrimo = false
	} else {
		for i := 2; i*i <= n; i++ {
			if n%i == 0 {
				isPrimo = false
				break
			}
		}
	}

	if isPrimo {
		fmt.Printf("%d é primo.\n", n)
	} else {
		fmt.Printf("%d não é primo.\n", n)
	}
}

File: Haskell Script/ex2.hs
main :: IO ()
main = do
    n <- solicitarPositivo "Digite um número inteiro positivo: "
    if ehPrimo n
        then putStrLn $ show n ++ " é primo."
        else putStrLn $ show n ++ " não é primo."

ehPrimo :: Int -> Bool
ehPrimo 1 = False
ehPrimo n = null [x | x <- [2..n-1], n `mod` x == 0]

solicitarPositivo :: String -> IO Int
solicitarPositivo prompt = do
    putStrLn prompt
    n <- readLn :: IO Int
    if n <= 0
        then do
            putStrLn "Por favor, digite um número positivo."
            solicitarPositivo prompt
        else return n
